### PR TITLE
Revert Revert Migrate jobs to use --provider, --gcp-project, --gcp-zone, --gcp-cloud-sdk and off KUBEMARK_TESTS

### PR DESCRIPTION
**Pull request by fejta**
Reverts commit 70993dfeaf263feb20f10a6cf1a50e1eaf41e746.

/assign @krzyzacy @shyamjvs 

ref https://github.com/kubernetes/test-infra/issues/3375

### CODE REVIEW

File: jenkins/bootstrap_test.py [hunk 1]
```diff
@@ -2210,18 +2244,23 @@ def test_envs_no_export(self):
                     ('JENKINS_USE_SERVER_VERSION=', '--extract=gke'),
                     ('JENKINS_USE_GCI_VERSION=', '--extract=gci/FAMILY'),
                     ('JENKINS_USE_GCI_HEAD_IMAGE_FAMILY=', '--extract=gci/FAMILY'),
+                    ('KUBE_GCE_ZONE=', '--gcp-zone='),
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: also add a check for `'Zone'`

Comment (fejta) [inline jenkins/bootstrap_test.py]: Done

**Comment by krzyzacy** [comment]
/lgtm
cc @spxtr

**Comment by luxas** [comment]
This broke all sig-cluster-lifecycle jobs including release-blocking ones: https://k8s-testgrid.appspot.com/sig-cluster-lifecycle
Between the earlier revert 1 and this PR the tests were healthy: https://github.com/kubernetes/test-infra/pull/3561

cc @mikedanese @pipejakob @wojtek-t 

```
W0719 10:09:15.361] flag provided but not defined: -gcp-project
W0719 10:09:15.361] Usage of kubetest:
W0719 10:09:15.362]   -build
W0719 10:09:15.362]         Rebuild k8s binaries, optionally forcing (release|quick|bazel) stategy
W0719 10:09:15.362]   -charts
W0719 10:09:15.362]         If true, run charts tests
W0719 10:09:15.362]   -check-leaked-resources
W0719 10:09:15.362]         Ensure project ends with the same resources
W0719 10:09:15.363]   -check-version-skew
W0719 10:09:15.363]         Verify client and server versions match (default true)
W0719 10:09:15.363]   -check_version_skew
W0719 10:09:15.364]         Verify client and server versions match (default true)
W0719 10:09:15.364]   -deployment string
W0719 10:09:15.364]         Choices: none/bash/kops/kubernetes-anywhere (default "bash")
W0719 10:09:15.364]   -down
W0719 10:09:15.364]         If true, tear down the cluster before exiting.
W0719 10:09:15.365]   -dump string
W0719 10:09:15.365]         If set, dump cluster logs to this location on test or cluster-up failure
W0719 10:09:15.365]   -extract value
W0719 10:09:15.365]         Extract k8s binaries from the specified release location
W0719 10:09:15.366]   -federation
W0719 10:09:15.366]         If true, start/tear down the federation control plane along with the clusters. To only start/tear down the federation control plane, specify --deploy=none
W0719 10:09:15.366]   -kops string
W0719 10:09:15.366]         (kops only) Path to the kops binary. Must be set for kops.
W0719 10:09:15.367]   -kops-admin-access string
W0719 10:09:15.367]         (kops only) If set, restrict apiserver access to this CIDR range.
W0719 10:09:15.367]   -kops-args string
W0719 10:09:15.367]         (kops only) Additional space-separated args to pass unvalidated to 'kops create cluster', e.g. '--kops-args="--dns private --node-size t2.micro"'
W0719 10:09:15.367]   -kops-cluster string
W0719 10:09:15.368]         (kops only) Cluster name. Must be set for kops.
W0719 10:09:15.368]   -kops-image string
W0719 10:09:15.368]         (kops only) Image (AMI) for nodes to use. (Defaults to kops default, a Debian image with a custom kubernetes kernel.)
W0719 10:09:15.368]   -kops-kubernetes-version string
W0719 10:09:15.369]         (kops only) If set, the version of Kubernetes to deploy (can be a URL to a GCS path where the release is stored) (Defaults to kops default, latest stable release.).
W0719 10:09:15.369]   -kops-nodes int
W0719 10:09:15.369]         (kops only) Number of nodes to create. (default 2)
W0719 10:09:15.369]   -kops-ssh-key string
W0719 10:09:15.370]         (kops only) Path to ssh key-pair for each node. (Defaults to $AWS_SSH_KEY or '~/.ssh/kube_aws_rsa'.)
W0719 10:09:15.370]   -kops-state string
W0719 10:09:15.370]         (kops only) s3:// path to kops state store. Must be set. (This flag defaults to $KOPS_STATE_STORE, and overrides it if set.)
W0719 10:09:15.370]   -kops-up-timeout duration
W0719 10:09:15.370]         (kops only) Time limit between 'kops config / kops update' and a response from the Kubernetes API. (default 20m0s)
W0719 10:09:15.371]   -kops-zones string
W0719 10:09:15.371]         (kops AWS only) AWS zones for kops deployment, comma delimited. (default "us-west-2a")
W0719 10:09:15.371]   -kubemark
W0719 10:09:15.371]         If true, run kubemark tests.
W0719 10:09:15.371]   -kubernetes-anywhere-cluster string
W0719 10:09:15.372]         (kubernetes-anywhere only) Cluster name. Must be set for kubernetes-anywhere.
W0719 10:09:15.372]   -kubernetes-anywhere-kubeadm-version string
W0719 10:09:15.372]         (kubernetes-anywhere only) Version of kubeadm to use, if phase2-provider is kubeadm. May be "stable" or a gs:// link to a custom build. (default "stable")
W0719 10:09:15.373]   -kubernetes-anywhere-kubernetes-version string
W0719 10:09:15.374]         (kubernetes-anywhere only) Version of Kubernetes to use (e.g. latest, stable, latest-1.6, 1.6.3, etc).
W0719 10:09:15.374]   -kubernetes-anywhere-path string
W0719 10:09:15.374]         (kubernetes-anywhere only) Path to the kubernetes-anywhere directory. Must be set for kubernetes-anywhere.
W0719 10:09:15.374]   -kubernetes-anywhere-phase2-provider string
W0719 10:09:15.374]         (kubernetes-anywhere only) Provider for phase2 bootstrapping. (Defaults to ignition). (default "ignition")
W0719 10:09:15.375]   -kubernetes-anywhere-up-timeout duration
W0719 10:09:15.375]         (kubernetes-anywhere only) Time limit between starting a cluster and making a successful call to the Kubernetes API. (default 20m0s)
W0719 10:09:15.375]   -multiple-federations
W0719 10:09:15.375]         If true, enable running multiple federation control planes in parallel
W0719 10:09:15.375]   -perf-tests
W0719 10:09:15.376]         If true, run tests from perf-tests repo.
W0719 10:09:15.376]   -publish string
W0719 10:09:15.376]         Publish version to the specified gs:// path on success
W0719 10:09:15.376]   -save string
W0719 10:09:15.376]         Save credentials to gs:// path on --up if set (or load from there if not --up)
W0719 10:09:15.376]   -skew
W0719 10:09:15.377]         If true, run tests in another version at ../kubernetes/hack/e2e.go
W0719 10:09:15.377]   -stage value
W0719 10:09:15.377]         Upload binaries to gs://bucket/devel/job-suffix if set
W0719 10:09:15.377]   -stage-suffix string
W0719 10:09:15.377]         Append suffix to staged version when set
W0719 10:09:15.378]   -test
W0719 10:09:15.378]         Run Ginkgo tests.
W0719 10:09:15.379]   -test_args string
W0719 10:09:15.379]         Space-separated list of arguments to pass to Ginkgo test runner.
W0719 10:09:15.379]   -timeout duration
W0719 10:09:15.380]         Terminate testing after the timeout duration (s/m/h)
W0719 10:09:15.380]   -up
W0719 10:09:15.380]         If true, start the the e2e cluster. If cluster is already up, recreate it.
W0719 10:09:15.380]   -upgrade_args string
W0719 10:09:15.381]         If set, run upgrade tests before other tests
W0719 10:09:15.381]   -v    If true, print all command output.
```

Apparently, the kubetest binary used in kubeadm tests are older than HEAD. Maybe we just should build a new `e2e-kubeadm` image from HEAD and bump https://github.com/kubernetes/test-infra/blob/master/prow/config.yaml#L779?

**Comment by luxas** [comment]
I think a forward-fix here is way better than reverting this a third and fourth time ;)

**Comment by luxas** [comment]
@krzyzacy I'm not sure if it's done by hand or not, but I guess not since things started to fail like this.

**Comment by luxas** [comment]
I do not have access to touch this anyway, so I have to rely on you @fejta or @pipejakob to do it ;)

**Comment by krzyzacy** [comment]
I'll bump the image

**Pull request merged**